### PR TITLE
Update vsix generation to use latest tools.

### PR DESCRIPTION
**Pull request by brettfo**

### CODE REVIEW

**Comment by heejaechang** [comment]
@brettfo is it your issue or infrastructure issue?

**Comment by brettfo** [comment]
@heejaechang Mine to update the tooling in this repo and ideally infrastructure's issue to do the insertion, but since there are other moving parts I'll handle this one end-to-end.

File: build.cmd [hunk 1]
```diff
@@ -389,7 +389,8 @@ if exist "%ProgramFiles(x86)%\Microsoft Visual Studio 12.0\common7\ide\devenv.ex
 if exist "%ProgramFiles%\Microsoft Visual Studio 12.0\common7\ide\devenv.exe" set VisualStudioVersion=12.0
 
 :vsversionset
-if '%VisualStudioVersion%' == '' echo Error: Could not find an installation of Visual Studio && goto :failure
+if "%VisualStudioVersion%" == "" echo Error: Could not find an installation of Visual Studio && goto :failure
+if "%VSSDK150INSTALL%" == "" (if "%BUILD_VS%" == "1" echo Error: Could not find an installation of Visual Studio 2015 VSSDK which is required for building VS components && goto :failure)
```
Comment (OmarTawfik) [inline build.cmd]: I thought with the move to VSSDK nuget, we shouldnet need this whole version check? we would never use the local VSSDK?

File: vsintegration/packages.config [hunk 2]
```diff
@@ -16,12 +16,14 @@
   <package id="Microsoft.VisualStudio.Threading" version="14.1.131" targetFramework="net46" />
   <package id="Microsoft.VisualStudio.CoreUtility" version="14.3.25407" targetFramework="net46" />
   <package id="Microsoft.VisualStudio.Shell.Design" version="14.3.25407" targetFramework="net46" />
-  <package id="Microsoft.VisualStudio.Utilities" version="14.3.25407" targetFramework="net46" />  
+  <package id="Microsoft.VisualStudio.Settings.15.0" version="15.0.25824-RC" targetFramework="net46" />
+  <package id="Microsoft.VisualStudio.Utilities" version="14.3.25407" targetFramework="net46" />
   <package id="Microsoft.VisualStudio.Language.StandardClassification" version="14.3.25407" targetFramework="net46" />
-  <package id="Microsoft.VisualStudio.Language.Intellisense" version="14.3.25407" targetFramework="net46" />  
+  <package id="Microsoft.VisualStudio.Language.Intellisense" version="14.3.25407" targetFramework="net46" />
```
Comment (OmarTawfik) [inline vsintegration/packages.config]: Do we need to update these references in any other `packages.config` in the rep?

**Comment by OmarTawfik** [comment]
Do we need to update the `.vsmanproj` to reference the json file produced out of the `.vsixproj` instead of the `.swixproj`?

File: build.cmd (hunk 1 above)
Comment (brettfo) [inline build.cmd]: Good point.  I'll remove this check.

**Comment by brettfo** [comment]
The generated `.json` files have the same name as the old ones so no update to the `.vsmanproj` is needed.

File: vsintegration/packages.config (hunk 2 above)
Comment (brettfo) [inline vsintegration/packages.config]: VS binding redirects will continue to take care of those until we have RTM packages that we can reference.

File: vsintegration/Vsix/VisualFSharpFull/Source.extension.vsixmanifest [hunk 3]
```diff
@@ -5,7 +5,9 @@
     <Identity Id="VisualFSharp" Version="15.4.1.0" Language="en-US" Publisher="Microsoft.VisualFSharpTools" />
     <DisplayName>Visual F# Tools</DisplayName>
     <Description xml:space="preserve">Deploy Visual F# Tools templates to Visual Studio</Description>
-    <ShortcutPath>..\CommonExtensions\Microsoft\FSharp</ShortcutPath>
+    <PackageId>Microsoft.FSharp.VSIX.Full.Core</PackageId>
+    <ExtensionInstallationRoot>CommonExtensions</ExtensionInstallationRoot>
+    <ExtensionInstallationFolder>Microsoft\FSharp</ExtensionInstallationFolder>
```
Comment (tinaschrepfer) [inline vsintegration/Vsix/VisualFSharpFull/Source.extension.vsixmanifest]: These VSIXes both go to Microsoft\FSharp, which is not allowed.  They need to go to different installation folders.

File: vsintegration/Vsix/VisualFSharpFull/Source.extension.vsixmanifest [hunk 4]
```diff
@@ -5,7 +5,9 @@
     <Identity Id="VisualFSharp" Version="15.4.1.0" Language="en-US" Publisher="Microsoft.VisualFSharpTools" />
     <DisplayName>Visual F# Tools</DisplayName>
     <Description xml:space="preserve">Deploy Visual F# Tools templates to Visual Studio</Description>
-    <ShortcutPath>..\CommonExtensions\Microsoft\FSharp</ShortcutPath>
+    <PackageId>Microsoft.FSharp.VSIX.Full.Core</PackageId>
+    <ExtensionInstallationRoot>CommonExtensions</ExtensionInstallationRoot>
```
Comment (tinaschrepfer) [inline vsintegration/Vsix/VisualFSharpFull/Source.extension.vsixmanifest]: These (ExtensionInstallationFolder and ExtensionInstallationRoot) should be added to the csproj as build properties, not the VSIX manifest.

File: vsintegration/Vsix/VisualFSharpWeb/Source.extension.vsixmanifest [hunk 5]
```diff
@@ -5,7 +5,9 @@
     <Identity Id="VisualFSharp" Version="15.4.1.0" Language="en-US" Publisher="Microsoft.VisualFSharpTools" />
     <DisplayName>Visual F# Tools</DisplayName>
     <Description xml:space="preserve">Deploy Visual F# Tools templates to Visual Studio</Description>
-    <ShortcutPath>..\CommonExtensions\Microsoft\FSharp</ShortcutPath>
+    <PackageId>Microsoft.FSharp.VSIX.Web.Core</PackageId>
+    <ExtensionInstallationRoot>CommonExtensions</ExtensionInstallationRoot>
+    <ExtensionInstallationFolder>Microsoft\FSharp</ExtensionInstallationFolder>
```
Comment (tinaschrepfer) [inline vsintegration/Vsix/VisualFSharpWeb/Source.extension.vsixmanifest]: See comments above.

File: vsintegration/Vsix/VisualFSharpFull/Source.extension.vsixmanifest (hunk 3 above)
Comment (brettfo) [inline vsintegration/Vsix/VisualFSharpFull/Source.extension.vsixmanifest]: These extensions will never be installed at the same time.

File: vsintegration/Vsix/VisualFSharpFull/Source.extension.vsixmanifest (hunk 4 above)
Comment (brettfo) [inline vsintegration/Vsix/VisualFSharpFull/Source.extension.vsixmanifest]: Thanks, done.

**Pull request merged**